lib/eco/incparser/error_recovery.py

```diff
@@ -71,10 +75,19 @@ def recover(self, error_node):
     def is_valid_iso_subtree(self, node, error_offset):
         if node in self.rejects:
             # Node has already been rejected as a candidate for recovery
+            # XXX node.isolated = False # reset isolations that didn't work?
```

Comment:
This is part of the above comment and might be the fix for the problem described above (if it is a problem that is). Maybe I should make that more clear.